### PR TITLE
[VoiceOver] Site Creation → Domains enhancements

### PR DESCRIPTION
**Pull request by shiki**
Refs #12127. 

This doesn't completely _fix_ #12127. I feel like there is still more we can do with the Domains step. But I'm not quite sure what those are. This PR hopefully makes the UI a bit easier to understand by VoiceOver users. 

### Announcing Suggestions Update

To help guide what needs to be done after entering something in the search field, I added a “Suggestions updated” announcement for VoiceOver. This idea was taken from [Android's implementation](https://github.com/wordpress-mobile/WordPress-Android/commit/990f0de44ea1702f87bc28c81e8bf1ac810f0fcd) which does the same thing.

### Create Site Button Visibility

By default, the _Create Site_ button is only shown when the user has selected a domain. My theory related to this is UI elements that are hidden and shown after an action is not ideal for VoiceOver users. There is usually no indication that something has happened in the UI. And it is less easy to understand how the UI works if a critical button is not there. 

In cases like this, we could:

a. Announce and focus the button when it is visible like the Notices (https://github.com/wordpress-mobile/WordPress-iOS/pull/12873). 
b. Make the button visible but disabled like the Insert button in Aztec (https://github.com/wordpress-mobile/WordPress-iOS/pull/13050). 

I decided to proceed with (b) as changing the focus (a) can be a jarring experience if they're just moving around on the domains list. 

### Labels and Hints

I also added various labels and hints:

- The domain cell is now a button (154987b16cd0f76917321036e9272cce7d94f5bd) and has a hint (254011360de1b3604c0c4f123bf224e9aa47c7e0)
- Added a hint to the search (839e92ee8ededdc4a6dd71648b40e5935b676d8b). I wasn't sure about this one but I hope it helps. 

## Outstanding Issues

I noticed that the domain search field gets disabled after you pick a domain. I think this is a bug and may have been unintentionally caused by https://github.com/wordpress-mobile/WordPress-iOS/pull/11739. I'll create a separate issue for it. 

## Testing

1. Turn VoiceOver on *before* tapping on the Add site button.
2. Tap on My Sites → Add button
3. Go through the steps until Step 3 of 3 (Domains). 
4. Confirm that:
    a. The Create Site button is always visible. It is enabled and disabled instead. 
    b. A “Suggestions updated” is announced after you finish typing something in the search field and waited for a bit until the search results are there. 
    c. The labels and hints described above are announced. 

## Submitter Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.
- [x] If it's feasible, I have added unit tests.

### CODE REVIEW

**Comment by shiki** [comment]
Hi @etoledom! Do you have time to review this? I'd really love to hear what you think of these changes. 🙂

File: WordPress/Classes/ViewRelated/Site Creation/WebAddress/WebAddressWizardContent.swift [hunk 1]
```diff
@@ -454,6 +460,29 @@ final class WebAddressWizardContent: UIViewController {
...
+
+    private func toggleBottomToolbar(enabled: Bool) {
+        createSite.isEnabled = enabled
+
+        if enabled {
+            tableViewOffsetCoordinator?.showBottomToolbar()
+        } else {
+            if !isBottomToolbarAlwaysVisible {
+                tableViewOffsetCoordinator?.hideBottomToolbar()
+            }
```
[the remark this answers is not in the document]
Comment (etoledom) [reply WordPress/Classes/ViewRelated/Site Creation/WebAddress/WebAddressWizardContent.swift]: >  it's probably better if we fix all of the issues together as they're all intertwined. 

💯 - Let's keep moving forward and not block this PR ✅

**Comment by pinarol** [comment]
Hey there! I've moved this to 14.0 since 13.9 has been cut. If you want this to make it to 13.9, please feel free to ping me.

**Comment by shiki** [comment]
Thank you, @etoledom! 🙏

**Comment by pinarol** [comment]
very sorry for the inconsistency, this one made it to 13.9, so I am fixing the milestone.

**Pull request merged**